### PR TITLE
[1LP][RFR] Move FQDN setting into Appliance.set_resolvable_hostname

### PR DESCRIPTION
**Pull request by mshriver**
Allows for calling in custom_configure as-is currently done, and for calling separately during deployment of a snapshotted configured image in sprout

### CODE REVIEW

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -2480,6 +2480,42 @@ def is_ip_available():
         appliance.name = name or cls._default_name
         return appliance
 
+    def set_resolvable_hostname(self):
+        """Try to lookup the hostname based on self.hostname, which is generally an IP
+
+        If a hostname is resolved for the IP, its used to set the hostname via appliance_cli
+        If an IP is resolved for self.hostname, no action is taken
```
Comment (izapolsk) [inline cfme/utils/appliance/__init__.py]: please add a note that it is relevant to appliances hosted on openstack.

Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: done

File: sprout/appliances/tasks.py [hunk 2]
```diff
@@ -1627,6 +1628,12 @@ def appliance_yum_update(self, appliance_id):
     appliance.ipapp.update_rhel(reboot=False)
 
 
+@singleton_task()
+def appliance_set_hostname(appliance_id):
+    appliance = Appliance.objects.get(id=appliance_id)
+    appliance.ipapp.set_resolvable_hostname(reboot=False)
```
Comment (izapolsk) [inline sprout/appliances/tasks.py]: REQUIRED: reboot=False ?

File: sprout/appliances/tasks.py [hunk 3]
```diff
@@ -1627,6 +1628,12 @@ def appliance_yum_update(self, appliance_id):
     appliance.ipapp.update_rhel(reboot=False)
 
 
+@singleton_task()
+def appliance_set_hostname(appliance_id):
+    appliance = Appliance.objects.get(id=appliance_id)
```
Comment (izapolsk) [inline sprout/appliances/tasks.py]: REQUIRED: I believe you need to to check whether this appliance is hosted on openstack before doing changes.

File: sprout/appliances/tasks.py (hunk 2 above)
Comment (mshriver) [reply sprout/appliances/tasks.py]: copypasta, can't believe I missed that.

File: sprout/appliances/tasks.py (hunk 3 above)
Comment (mshriver) [reply sprout/appliances/tasks.py]: done.

File: sprout/appliances/tasks.py [hunk 4]
```diff
@@ -1627,6 +1628,13 @@ def appliance_yum_update(self, appliance_id):
...
+    if appliance.provider.provider_data.get('type', None) == 'openstack':
+        appliance.ipapp.set_resolvable_hostname()
```
Comment (izapolsk) [inline sprout/appliances/tasks.py]: REQUIRED: are you sure it will work well w/o callback ?

Comment (mshriver) [reply sprout/appliances/tasks.py]: it will now that I wrapped `set_resolvable_hostname` with `logger_wrap` decorator. :)

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
@izapolsk after testing this locally (thanks for your help in setup!!!) I think this is a working solution to setting hostnames.

I found that appliance_console_cli isn't cleaning up hostnames, so there is either a BZ to be fixed or workaround to be implemented in seal_for_templatizing. I don't think that's a blocker to merging this as-is and updating production sprout so that openstack instances will have resolvable hostnames.

**Pull request merged**